### DISCUSSION TITLE
Is possible to used both types on rust and on frontend?

### DISCUSSION THREAD

**Discussion by codeitlikemiley** [tauri-apps/tauri#10994] [Q&A]
I see that on the documentation that most of the code have to be re-written twice...
e.g. 
if we define a struct of User 

if we broadcast an Event with User

on typescript we need to redefine those types

as such its kinda pointless to do it manually

I know its possible to compiled on a shared and shared types library  types e.g. crux 

i just dont know if someone already done it 

if so please share it thanks...

if not i will be the one to create that functionality myself :)

**Comment by FabianLars** [comment]
Check out this comment: https://github.com/tauri-apps/tauri/issues/1514#issuecomment-1619337304 - it should still be accurate.
